### ISSUE TITLE
Nested tab stops now expand incorrectly

### ISSUE THREAD

**Issue by isiahmeadows** [atom/snippets#266]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

[Description of the issue]

### Steps to Reproduce

1. Add a snippet with the body `'() => ${1:\\{$2\\}}'`
1. Open a new file
1. Change to a language where that snippet will be in scope
1. Start to type it and hit tab to expand it

**Expected behavior:** [What you expect to happen]

It to expand like this with the brackets selected:

```js
() => {}
```

**Actual behavior:** [What actually happens]

It expands like this, with the brackets selected:

```js
() => {[object Object]}
```

**Reproduces how often:** [What percentage of the time does it reproduce?]

100% of the time

### Versions

You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

Atom v1.25.0 on macOS 64-bit. This bug did not exist in v1.24.1.

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by nemoDreamer** [comment]
I have a bunch of snippets where a group references another group:

```
'.source.js':
    'Console Log':
        'prefix': 'cl'
        'body': '$3console.${2:log}($1);$0'
    'Console Log (verbose)':
        'prefix': 'cll'
        'body': "$4console.${3:log}('${2:$1}', $1);$0"
    'Console Log (labeled, verbose)':
        'prefix': 'clll'
        'body': "$5console.${4:log}('$1: ${3:$2}', $2);$0"
```

Typing `cll ⇥` and typing something would result in 2 cursors:

```
console.log("myMysteryHash|", myMysteryHash|);
```

But now, it only does

```
console.log("[object Object]", myMysteryHash|);
```

---

I do not see the `${2:$1}` pattern mentioned anywhere in the snippet docs. Was I just taking advantage of something I shouldn't have?

Could there be a `backreference` format instead, like `${2:\\1}`?

**Comment by joefitzgerald** [comment]
This is definitely an issue; it's affecting [`go-plus`](https://atom.io/packages/go-plus) autocomplete snippet expansion:

![kapture 2018-04-09 at 16 00 06](https://user-images.githubusercontent.com/744740/38525420-77e22848-3c0f-11e8-8ee3-6a7cb07ada2c.gif)

![screen shot 2018-04-09 at 4 02 13 pm](https://user-images.githubusercontent.com/744740/38525424-7cd23b36-3c0f-11e8-967a-8cd44603e078.png)

**Comment by savetheclocktower** [comment]
Oddly enough, this usage had no coverage in the tests. #267 ought to fix it.

**Comment by rsese** [comment]
Thanks for the report @isiahmeadows, just noting that I can reproduce on macOS 10.12.6 with 1.26.0-beta1 but it's not reproducible on 1.24.1 like you mentioned.

**Comment by isiahmeadows** [comment]
@rsese Thanks! (It kind of didn't help that it broke some of my most frequently used snippets - I noticed within minutes of upgrading and about 36 hours from [when it was first released](https://github.com/atom/atom/releases/tag/v1.25.0).)

**Comment by jasonrudolph** [comment]
Fixed in #267
